### PR TITLE
Exclude config blobs on demand for Job and Build

### PR DESCRIPTION
**Pull request by meatballhat**
to cut down on database and network load.

### CODE REVIEW

File: lib/travis/services/find_build.rb [hunk 1]
```diff
@@ -31,8 +32,16 @@ def all_resources
...
         end
 
-        def result
-          @result ||= scope(:build).find_by_id(params[:id])
+        def result(options = {})
+          @result ||= load_result(options)
+        end
+
+        def load_result(options = {})
+          columns = scope(:build).column_names
```
Comment (joshk) [inline lib/travis/services/find_build.rb]: smart!

File: lib/travis/services/find_build.rb [hunk 2]
```diff
@@ -31,8 +32,16 @@ def all_resources
...
+          columns -= %w(config) if options[:exclude_config]
+          scope(:build).select(columns).find_by_id(params[:id]).tap do |res|
+            res.config = {} if options[:exclude_config]
```
Comment (joshk) [inline lib/travis/services/find_build.rb]: why do you do this?

Comment (joshk) [inline lib/travis/services/find_build.rb]: why not leave it be nil?

Comment (meatballhat) [inline lib/travis/services/find_build.rb]: By default it's not merely `nil`, but an unresolved relation.  Other parts of the code (somewhere!?!) want to treat it as a hash to default in `language` and `os`, but without making it a hash first we end up with ActiveRecord explosions.

File: lib/travis/services/find_build.rb (hunk 1 above)
Comment (meatballhat) [inline lib/travis/services/find_build.rb]: :trollface: http://stackoverflow.com/questions/17431028/activerecord-select-except-columns/17431177#17431177

File: lib/travis/services/find_build.rb (hunk 2 above)
Comment (joshk) [inline lib/travis/services/find_build.rb]: ok, we can go ahead with this if you prefer setting to `{}` vs `nil`

**Pull request merged**